crates/matrix-sdk/src/room/timeline/tests.rs

```diff
@@ -794,6 +786,60 @@ fn make_message_event<C: MessageLikeEventContent>(sender: &UserId, content: C) -
     })
 }
 
+fn make_state_event<C: StateEventContent>(
+    sender: &UserId,
+    state_key: &str,
+    content: C,
+    prev_content: Option<C>,
+) -> JsonValue {
+    let unsigned = if let Some(prev_content) = prev_content {
+        json!({ "prev_content": prev_content })
+    } else {
+        json!({})
+    };
+
+    json!({
+        "type": content.event_type(),
+        "state_key": state_key,
+        "content": content,
+        "event_id": EventId::new(server_name!("dummy.server")),
+        "sender": sender,
+        "origin_server_ts": next_server_ts(),
+        "unsigned": unsigned,
+    })
+}
+
+fn make_redacted_state_event<C: RedactedStateEventContent>(
+    sender: &UserId,
+    state_key: &str,
+    content: C,
+) -> JsonValue {
+    let event_id = &EventId::new(server_name!("dummy.server"));
+
+    json!({
+        "type": content.event_type(),
+        "state_key": state_key,
+        "content": content,
+        "event_id": event_id,
+        "sender": sender,
+        "origin_server_ts": next_server_ts(),
+        "unsigned": make_redacted_unsigned(sender, event_id),
+    })
+}
+
+fn make_redacted_unsigned(sender: &UserId, redacts: &EventId) -> JsonValue {
+    json!({
+        "redacted_because": {
+            "content": {},
+            "redacts": redacts,
```

Comment:
Because the `redacted_because` event can also be itself redacted and we would never read this field anyways, I think we can remove this here.